src/librustc_data_structures/slice_ext.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+pub trait SliceExt<T> {
+    type Item;
+
+    fn pick2(&mut self, a: usize, b: usize) -> (&mut T, &mut T);
```

Comment:
Either:

- This should be `&mut Self::Item` and there is no need for the `T` parameter

Or:

- Remove the `type Item`